ts/spi/provider_frame.ts

```diff
@@ -221,7 +229,7 @@ export class ProviderFrame {
       options: CredentialHintOptions) {
     console.info('Handling hint request');
 
-    let hints = await this.getHints(options);
+    let hints = await this.race([], this.getHints(options));
```

Comment:
You cannot only race here. The whole function has to be cancellable (as it is more likely that the need to cancel is after the UI is displayed, i.e. below when the interaction provider is called).

Didn't we discuss to race the cancellable promise at a higher level? (maybe in the `registerListeners`?)